feat(ref): add a yaml section (#196)
go-vela/docs: content/reference/yaml/stages.md

---
title: "Stages"
linkTitle: "Stages"
weight: 7
description: >
  YAML stages for stages block
---

> The stages tag is intended to be used to one to many sets of steps tasks in parallel.

Reviewer: This sentence reads weird to me

Not in love with my suggestion either 🤷 
```suggestion
The stages tag is intended to be used to parallelize one-to-many sets of step tasks.
```

# to run two stages with one step task in parallel.
stages:
  test:
    hello:
      - name: Hello World
        image: alpine:latest
        commands:
          - echo "Hello, Vela User"

  welcome:
    steps:
      - name: Welcome to Vela
        image: alpine:latest
        commands:
          - echo "Welcome to Vela!"
```

## Tags

| Tag     | Required | Type     | Description                                               |
|---------|----------|----------|-----------------------------------------------------------|
| `name`  | Y        | string   | Unique identifier for the stage in the pipeline           |
| `steps` | Y        | []string | Sequential execution instructions for the stage           |
| `needs` | N        | struct   | Stages that must complete before starting the current one |

### Usage

#### The `name:` tag

```yaml
---
stages:
    # Unique identifier for the stage in the pipeline.
    welcome:
```

#### The `steps:` tag

```yaml
---
stages:
    # Unique identifier for the stage in the pipeline.
    welcome:
      # Sequential execution instructions for the stage.
      steps:
```

#### The `steps:` tag

{{% alert title="Tip:" color="info" %}}

> To see how all steps tags, see the [step tags documentation](/docs/reference/yaml/steps/#tags)

Reviewer: Reads weird to me
```suggestion
For more details on all steps tags, see the [step tags documentation](/docs/reference/yaml/steps/#tags)
```
